assemble.go

```diff
@@ -162,19 +259,32 @@ func AssembleFile(ctx context.Context, name string, idx Index, s Store, seeds []
 		})
 	}
 
-	// Let the sequencer break up the index into segments, feed the workers, and
-	// stop if there are any errors
+	// Let the sequencer break up the index into segments, create and validate a plan,
+	// feed the workers, and stop if there are any errors
 	seq := NewSeedSequencer(idx, seeds...)
-loop:
+	plan := seq.Plan()
 	for {
-		chunks, from, done := seq.Next()
+		if err := validatePlan(ctx, plan); err != nil {
+			// This plan has at least one invalid seed
+			if options.InvalidSeedAction == InvalidSeedActionBailOut {
+				return stats, err
+			}
+			// Skip the invalid seed and try again
+			Log.WithError(err).Info("Unable to use one of the chosen seeds, skipping it")
+			seq.Rewind()
+			plan = seq.Plan()
+			continue
+		}
+		// Found a valid plan
+		break
+	}
+
+loop:
+	for _, candidates := range plan {
```

Comment:
Wrt naming, it's not really a "candidate" at this point anymore. Perhaps "segment" would be better?